File: python/courtlistener/api/fjc_integrated_database.py
"""
FJC Integrated Database API module for CourtListener SDK.
"""

from typing import Dict, Any, Optional, List
from .base import BaseAPI


class FJCIntegratedDatabaseAPI(BaseAPI):
    """API for accessing FJC integrated database."""
    
    def __init__(self, client):
        super().__init__(client)
        self.endpoint = "fjc-integrated-database/"
    
    def list(self, **kwargs) -> Dict[str, Any]:
        """
        List FJC integrated database records.
        
        Args:
            **kwargs: Query parameters
        
        Returns:
            API response with paginated list of FJC integrated database records
        """
        return self.client.get(self.endpoint, params=kwargs)
    
    def get(self, record_id: int) -> Dict[str, Any]:
        """
        Get a specific FJC integrated database record by ID.
        
        Args:
            record_id: FJC integrated database record ID
        
        Returns:
            FJC integrated database record data
        """
        return self.client.get(f"{self.endpoint}{record_id}/")
    
    def paginate(self, **kwargs):
        """
        Get paginated FJC integrated database records.
        
        Args:
            **kwargs: Query parameters
        
        Returns:
            PageIterator for FJC integrated database records
        """
        return self.client.paginate(self.endpoint, params=kwargs)

File: php/src/CourtListener/Api/FjcIntegratedDatabase.php
<?php

namespace CourtListener\Api;

use CourtListener\CourtListenerClient;
use CourtListener\Exceptions\CourtListenerException;

/**
 * FjcIntegratedDatabase API endpoint
 */
class FjcIntegratedDatabase extends BaseApi
{
    protected string $endpoint = 'fjc-integrated-database/';

    /**
     * Constructor
     *
     * @param CourtListenerClient $client
     */
    public function __construct(CourtListenerClient $client)
    {
        parent::__construct($client);
    }

    /**
     * List fjc integrated database with advanced filtering
     *
     * @param array $params Query parameters
     * @return array
     * @throws CourtListenerException
     */
    public function listFjcIntegratedDatabase(array $params = []) {
        return $this->list($params);
    }

    /**
     * Get a specific fjc integrated database by ID
     *
     * @param int|string $id FjcIntegratedDatabase ID
     * @param array $params Query parameters
     * @return array
     * @throws CourtListenerException
     */
    public function getFjcIntegratedDatabase($id, array $params = []) {
        return $this->get($id, $params);
    }

    /**
     * Search fjc integrated database
     *
     * @param array $params Search parameters
     * @return array
     * @throws CourtListenerException
     */
    public function searchFjcIntegratedDatabase(array $params = []) {
        return $this->search($params);
    }
}
